src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -449,39 +449,55 @@ private static FileStream OpenStream(int fd, FileAccess access)
         /// </remarks>
         private static void ParseArgumentsIntoList(string arguments, List<string> results)
         {
-            var currentArgument = new StringBuilder();
-            bool inQuotes = false;
-
             // Iterate through all of the characters in the argument string.
             for (int i = 0; i < arguments.Length; i++)
             {
+                while (i < arguments.Length && (arguments[i] == ' ' || arguments[i] == '\t'))
+                    i++;
+
+                if (i == arguments.Length)
+                    break;
+
+                results.Add(GetNextArgument(arguments, ref i));
+            }
+        }
+
+        private static string GetNextArgument(string arguments, ref int i)
+        {
+            var currentArgument = StringBuilderCache.Acquire();
+            bool inQuotes = false;
+
+            while (i < arguments.Length)
+            {
                 // From the current position, iterate through contiguous backslashes.
                 int backslashCount = 0;
-                for (; i < arguments.Length && arguments[i] == '\\'; i++, backslashCount++) ;
+                while (i < arguments.Length && arguments[i] == '\\')
+                {
+                    i++;
+                    backslashCount++;
+                }
+
                 if (backslashCount > 0)
                 {
                     if (i >= arguments.Length || arguments[i] != '"')
                     {
                         // Backslashes not followed by a double quote:
                         // they should all be treated as literal backslashes.
                         currentArgument.Append('\\', backslashCount);
-                        i--;
                     }
                     else
                     {
                         // Backslashes followed by a double quote:
                         // - Output a literal slash for each complete pair of slashes
                         // - If one remains, use it to make the subsequent quote a literal.
                         currentArgument.Append('\\', backslashCount / 2);
-                        if (backslashCount % 2 == 0)
-                        {
-                            i--;
-                        }
-                        else
+                        if (backslashCount % 2 != 0)
                         {
                             currentArgument.Append('"');
+                            i++;
                         }
                     }
+
                     continue;
                 }
 
```

Comment:
Also, if you autocrement i here: i.e.

   "char c = arguments[i++]"

you can eliminate some "i++"'s below. And it's clearer to do the index increment at the same time you're consuming the character.

(Note: This might imply adding a index decrement on the "break;" instruction below but in fact, you don't need that.)